app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala

```diff
@@ -0,0 +1,178 @@
+    applicationService: ApplicationService
+  )(implicit override val ec: ExecutionContext
+  ) extends AbstractApplicationController(strideAuthorisationService, mcc, errorHandler) with GatekeeperRoleWithApplicationActions with ApplicationLogger {
+  import TermsOfUseHistoryController._
+
+  def page(applicationId: ApplicationId) = loggedInThruStrideWithApplication(applicationId) { implicit request =>
+    def deriveSubmissionStatusDisplayName(status: Submission.Status): String = {
+      status match {
+        case s: Answering                    => "In progress"
+        case s: Created                      => "In progress"
+        case s: Declined                     => "Declined"
+        case s: Failed                       => "Failed"
+        case s: Granted                      => "Terms of use V2"
+        case s: GrantedWithWarnings          => "Terms of use V2 with warnings"
+        case s: PendingResponsibleIndividual => "Pending responsible individual"
+        case s: Submitted                    => "Submitted"
+        case s: Warnings                     => "Warnings"
+      }
+    }
+
+    def deriveSubmissionStatusDescription(status: Submission.Status): String = {
+      status match {
+        case s: Answering                    => "The submission was started"
+        case s: Created                      => "The submission was started"
+        case s: Declined                     => s"${s.name} declined that the application complies with the terms of use V2"
+        case s: Failed                       => s"${s.name} submitted the terms of use checklist.  The application did not comply with version 2 of the terms of use."
+        case s: Granted                      => s"${s.name} accepted that the application complies with the terms of use V2"
+        case s: GrantedWithWarnings          => s"${s.name} accepted that the application complies with the terms of use V2 with warnings"
+        case s: PendingResponsibleIndividual => "The submission is waiting for the responsible individual to accept the terms of use"
+        case s: Submitted                    => "The submission was submitted"
+        case s: Warnings                     => s"${s.name} submitted the terms of use checklist.  The application did not comply with version 2 of the terms of use."
+      }
+    }
+
+    def deriveSubmissionStatusDetail(status: Submission.Status): Option[String] = {
+      status match {
+        case s: Answering                    => None
+        case s: Created                      => None
+        case s: Declined                     => Some(s.reasons)
+        case s: Failed                       => None
+        case s: Granted                      => None
+        case s: GrantedWithWarnings          => Some(s.warnings)
+        case s: PendingResponsibleIndividual => None
+        case s: Submitted                    => None
+        case s: Warnings                     => None
+      }
+    }
```

Comment:
I was going to suggest munging these into one method and returning a tuple3 instead of having three methods each returning a String but... I am not sure if I like that idea the more I think about it.

Reply:
No, I originally had one method that returned the whole model, but it was big and nasty.